### PR TITLE
Vector for Kafka

### PR DESCRIPTION
**Pull request by gingerwizard**

### CODE REVIEW

File: docs/integrations/kafka/kafka-vector.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+Vector utilizes [sources](https://vector.dev/docs/about/concepts/#sources) for retrieving data through a push or pull model. [Sinks](https://vector.dev/docs/about/concepts/#sinks) meanwhile provide a destination for events. We, therefore, utilize the Kafka source and ClickHouse sink. Note that whilst Kafka is supported as a Sink, a ClickHouse source is not available. Vector is as a result not appropriate for users wishing to transfer data to Kafka from ClickHouse.
+
+ Vector also supports the [transformation](https://vector.dev/docs/reference/configuration/transforms/) of data. This is beyond the scope of this guide. The user is referred to the Vector documentation should they need this on their dataset.
+
+ Note that the current implementation of the ClickHouse sink utilizes the HTTP interface. The ClickHouse source does not support the use of a JSON schema at this time. Data must be published to Kafka in either plain JSON format or as Strings.
```
Comment (Ryado) [inline docs/integrations/kafka/kafka-vector.md:17]: > The ClickHouse source does not support the use of a JSON schema at this time. Data must be published to Kafka in either plain JSON format or as Strings.

What do we mean by the `ClickHouse source` in that context ? should it be `sink` or is it during the initial dataset generation? if yes, then we can leave out this bit as the reader can be confused by the "source" terminology also used by vector

File: docs/integrations/kafka/kafka-vector.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    assignee LowCardinality(String),
+    assignees Array(LowCardinality(String)),
+    closed_at DateTime,
+    merged_at DateTime,
+    merge_commit_sha String,
+    requested_reviewers Array(LowCardinality(String)),
+    merged_by LowCardinality(String),
+    review_comments UInt32,
+    member_login LowCardinality(String)
+) ENGINE = MergeTree ORDER BY (event_type, repo_name, created_at);
```
Comment (Ryado) [inline docs/integrations/kafka/kafka-vector.md:62]: It's better to use ReplicatedMergeTree ?

File: docs/integrations/kafka/kafka-vector.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+type = "kafka"
+auto_offset_reset = "smallest"
+bootstrap_servers = "<kafka_host>:<kafka_port>"
+group_id = "vector"
+topics = [ "github" ]
+tls.enabled = true
+sasl.enabled = true
+sasl.mechanism = "PLAIN"
+sasl.username = "BYYFAAU2CFOPWRGG"
+sasl.password = "cS+yZfr6QfmWgT/fBVN5aGUEZdC0Y2DnTquWG6lzDOL2JNkPSL+0WIKcXWylIeXe"
```
Comment (Ryado) [inline docs/integrations/kafka/kafka-vector.md:79]: We can replace sasl credentials with placeholders
`<username>`
`<password>`

**Review by Ryado — COMMENTED** [review]
LGTM overall, added minor comments

File: docs/integrations/kafka/kafka-vector.md (hunk 1 above)
Comment (gingerwizard) [reply docs/integrations/kafka/kafka-vector.md:17]: i mean sink

File: docs/integrations/kafka/kafka-vector.md (hunk 3 above)
Comment (gingerwizard) [reply docs/integrations/kafka/kafka-vector.md:79]: ahhhh nice catch

File: docs/integrations/kafka/kafka-vector.md (hunk 2 above)
Comment (Ryado) [reply docs/integrations/kafka/kafka-vector.md:62]: 👍

**Pull request merged**